src/coprocessor/codec/mysql/time/mod.rs

```diff
@@ -184,37 +186,68 @@ fn split_ymd_with_frac_as_hms(
     Ok((year, month, day, hour, minute, sec))
 }
 
+#[derive(Clone, Debug, Copy, PartialEq)]
+pub enum TimeType {
+    Date,
+    DateTime,
+    Timestamp,
+}
+
+impl Into<FieldTypeTp> for TimeType {
+    fn into(self) -> FieldTypeTp {
+        match self {
+            TimeType::Date => FieldTypeTp::Date,
+            TimeType::DateTime => FieldTypeTp::DateTime,
+            TimeType::Timestamp => FieldTypeTp::Timestamp,
+        }
+    }
+}
+
+impl TryFrom<FieldTypeTp> for TimeType {
+    type Error = Error;
+
+    fn try_from(value: FieldTypeTp) -> Result<Self> {
+        match value {
+            FieldTypeTp::Date => Ok(TimeType::Date),
+            FieldTypeTp::DateTime => Ok(TimeType::DateTime),
+            FieldTypeTp::Timestamp => Ok(TimeType::Timestamp),
+            FieldTypeTp::Unspecified => Ok(TimeType::DateTime), // FIXME: We should forbid this
```

Comment:
Is this behavior the same as before?

Reply:
Yes, it is just not rigorous because other FieldType should be never evaluated to a `Time`.